addOns/selenium/src/main/java/org/zaproxy/zap/extension/selenium/AbstractMultipleOptionsTablePanel.java

```diff
@@ -0,0 +1,119 @@
+ * available with that constructor visible.
+ */
+public abstract class AbstractMultipleOptionsTablePanel<E extends EnableableInterface>
+        extends AbstractMultipleOptionsBaseTablePanel<E> {
+
+    private static final long serialVersionUID = 1L;
+    private static final String ENABLE_ALL_BUTTON_LABEL =
+            Constant.messages.getString("multiple.options.panel.enableAll.button.label");
+    private static final String DISABLE_ALL_BUTTON_LABEL =
+            Constant.messages.getString("multiple.options.panel.disableAll.button.label");
+
+    private JButton enableAllButton;
+    private JButton disableAllButton;
+
+    public AbstractMultipleOptionsTablePanel(AbstractMultipleOptionsTableModel<E> model) {
+        this(model, true);
+    }
+
+    public AbstractMultipleOptionsTablePanel(
+            AbstractMultipleOptionsTableModel<E> model, boolean allowModification) {
+        super(model, allowModification);
+
+        final boolean buttonsEnabled = getModel().getRowCount() > 0;
+
+        enableAllButton = new JButton(ENABLE_ALL_BUTTON_LABEL);
+        enableAllButton.setEnabled(buttonsEnabled);
+        enableAllButton.addActionListener(
+                new ActionListener() {
+
+                    @Override
+                    public void actionPerformed(ActionEvent e) {
+                        ((AbstractMultipleOptionsTableModel<?>) getMultipleOptionsModel())
+                                .setAllEnabled(true);
+                    }
+                });
+
+        disableAllButton = new JButton(DISABLE_ALL_BUTTON_LABEL);
+        disableAllButton.setEnabled(buttonsEnabled);
+        disableAllButton.addActionListener(
+                new ActionListener() {
+
+                    @Override
+                    public void actionPerformed(ActionEvent e) {
+                        ((AbstractMultipleOptionsTableModel<?>) getMultipleOptionsModel())
+                                .setAllEnabled(false);
+                    }
+                });
+
+        addButtonSpacer();
+
+        addButton(enableAllButton);
+        addButton(disableAllButton);
+
+        getModel()
+                .addTableModelListener(
+                        new TableModelListener() {
+
+                            @Override
+                            public void tableChanged(TableModelEvent e) {
+                                if (TableModelEvent.ALL_COLUMNS == e.getColumn()
+                                        || TableModelEvent.INSERT == e.getType()
+                                        || TableModelEvent.DELETE == e.getType()) {
+                                    boolean enabled = getModel().getRowCount() > 0;
+
+                                    enableAllButton.setEnabled(enabled);
+                                    disableAllButton.setEnabled(enabled);
+                                }
+                            }
+                        });
```

Comment:
Not sure if it's practical here, but lambda?

Reply:
Again, this is a copy of a core class which will be replaced in 2.12.0